mlflow/pyfunc/__init__.py

```diff
@@ -1054,6 +1179,42 @@ def spark_udf(spark, model_uri, result_type="double", env_manager=_EnvManager.LO
 
     model_metadata = Model.load(os.path.join(local_model_path, MLMODEL_FILE_NAME))
 
+    model_output_schema = model_metadata.get_output_schema()
+
+    if result_type is None:
+        if model_output_schema is None:
```

Comment:
```suggestion

    if result_type is None:
        if model_output_schema := model_metadata.get_output_schema():
            ...
        else:
            # warning
```

We don't need to fetch schema if `result_type` is explicitly specified.